File: Algorithm/Leetcode/0000_0100/0011_0020/0016.cpp
//1313. Decompress Run-Length Encoded List

class Solution {
public:
    vector<int> decompressRLElist(vector<int>& nums) {
        std::vector<int> output;
        int num = 0;
        for(size_t i = 0; i < nums.size();  i += 2)
        {
            num +=  nums[i];
        }
        output.reserve(num);
        
		for (size_t i = 0; i < nums.size();  i += 2)
		{
			output.insert(output.end(), nums[i], nums[i + 1]);
		}
		return output;
    }
};

File: Algorithm/Leetcode/0000_0100/0011_0020/0016.py
# 1313. Decompress Run-Length Encoded List

"""https://leetcode.com/problems/decompress-run-length-encoded-list/discuss/477055/Python-1-Line-with-Intuition"""

class Solution:
    def decompressRLElist(self, nums: List[int]) -> List[int]:
        return [x for a, b in zip(nums[::2], nums[1::2]) for x in [b] * a]